gamatrix-gog.py

```diff
@@ -103,7 +130,102 @@ def init_opts():
     }
 
 
-def build_config(args):
+def build_config(args: Dict[str, Any]) -> Dict[str, Any]:
+    """Returns a config dict created from the config file and
+    command-line arguments, with the latter taking precedence
+    """
+    if args.get("--config-file", None) is not None:
+        yaml = YAML(typ="safe")
+        with open(args["--config-file"], "r") as config_file:
+            config = yaml.load(config_file)
+    else:
+        # We didn't get a config file, so populate from args
+        config = {}
+
+    # TODO: allow using both IDs and DBs (use one arg and detect if it's an int)
+    # TODO: should be able to use unambiguous partial names
+    if not args.get("<db>", []) and "users" not in config:
+        raise ValueError("You must use -u, have users in the config file, or list DBs")
+
+    # Command-line args override values in the config file
+    # TODO: maybe we can do this directly in argparse, or otherwise better
+
+    # This can't be given as an argument as it wouldn't make much sense;
+    #  provide a sane default if it's missing from the config file
+    if "db_path" not in config:
+        config["db_path"] = "."
+
+    config["all_games"] = args.get("--all-games", False)
+
+    config["include_single_player"] = args.get("--include-single-player", False)
+
+    if args.get("--server", True):  # Note that the --server opt is False unless present
```

Comment:
So you're defaulting to server mode? Not sure I quite understand how this works. If `--server` isn't used, then it defaults to `True`, so mode is set to `"server"`?